File: Domain/Recurrence.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain
{
    public class Recurrence
    {
        public Guid Id { get; set; }
        public bool Sunday { get; set; }
        public bool Monday { get; set; }
        public bool Tuesday { get; set; }
        public bool Wednesday { get; set; }
        public bool Thursday { get; set; }
        public bool Friday { get; set; }
        public bool Saturday { get; set; }
        public string Interval { get; set; }
        public string DayOfMonth { get; set; }
        public string WeekOfMonth { get; set; }
        public string WeekdayOfMonth { get; set; }
        public DateTime IntervalStart { get; set; }
        public DateTime IntervalEnd { get; set; }
        public bool IncludeWeekends { get; set; }
        public string DaysRepeating { get; set; }
        public string WeeksRepeating { get; set; }
        public string MonthsRepeating { get; set; }
        public string WeekInterval { get; set; }
        public string WeekendsIncluded { get; set; }
        public string WeeklyRepeatType { get; set; }
        public string MonthlyRepeatType { get; set; }
        public string MonthlyDayType { get; set; }
        public IEnumerable<Activity> Activities { get; set; }
        [NotMapped]
        public DateTime? ActivityStart { get; set; }
        [NotMapped]
        public DateTime? ActivityEnd { get; set; }

        public bool ArePropertiesEqual(Recurrence other)
        {
            bool isEqual = true;

            if (other == null)
                return false;

            isEqual &= Sunday == other.Sunday;
            isEqual &= Monday == other.Monday;
            isEqual &= Tuesday == other.Tuesday;
            isEqual &= Wednesday == other.Wednesday;
            isEqual &= Thursday == other.Thursday;
            isEqual &= Friday == other.Friday;
            isEqual &= Saturday == other.Saturday;
            isEqual &= string.Equals(Interval, other.Interval);
            isEqual &= string.Equals(DayOfMonth, other.DayOfMonth);
            isEqual &= string.Equals(WeekOfMonth, other.WeekOfMonth);
            isEqual &= string.Equals(WeekdayOfMonth, other.WeekdayOfMonth);
            isEqual &= IntervalStart == other.IntervalStart;
            isEqual &= IntervalEnd == other.IntervalEnd;
            isEqual &= IncludeWeekends == other.IncludeWeekends;
            isEqual &= string.Equals(DaysRepeating, other.DaysRepeating);
            isEqual &= string.Equals(WeeksRepeating, other.WeeksRepeating);
            isEqual &= string.Equals(MonthsRepeating, other.MonthsRepeating);
            isEqual &= string.Equals(WeekInterval, other.WeekInterval);
            isEqual &= string.Equals(WeekendsIncluded, other.WeekendsIncluded);
            isEqual &= string.Equals(WeeklyRepeatType, other.WeeklyRepeatType);
            isEqual &= string.Equals(MonthlyRepeatType, other.MonthlyRepeatType);
            isEqual &= string.Equals(MonthlyDayType, other.MonthlyDayType);

            return isEqual;
        }


    }
}

File: client-app/src/app/models/recurrence.ts
import { store } from "../stores/store"
import { v4 as uuid } from 'uuid';


export interface Recurrence{
    id: string,
    sunday: boolean
    monday: boolean
    tuesday: boolean
    wednesday: boolean
    thursday: boolean
    friday: boolean
    saturday: boolean
    interval: string
    dayOfMonth: string
    weekOfMonth: string
    weekdayOfMonth: string
    intervalStart: Date
    intervalEnd: Date
    includeWeekends: boolean
    daysRepeating: string
    weeksRepeating: string
    monthsRepeating: string
    weekInterval: string
    weekendsIncluded: string
    weeklyRepeatType: string
    monthlyRepeatType: string
    monthlyDayType: string
    activityStart? : Date
    activityEnd?: Date
}

export class Recurrence implements Recurrence{
    constructor(init?: RecurrenceFormValues){
      Object.assign(this, init);
    }
}

export class RecurrenceFormValues{
    id: string = uuid();
    sunday: boolean = false;
    monday: boolean = true;
    tuesday: boolean = false;
    wednesday: boolean = false;
    thursday: boolean = false;
    friday: boolean = false;
    saturday: boolean = false;
    interval: string = 'daily';
    dayOfMonth: string = '1';
    weekOfMonth: string = '1';
    weekdayOfMonth: string = '1'
    intervalStart: Date = new Date(new Date().setHours(0,0,0,0));
    intervalEnd: Date =  store.commonStore.addDays(new Date(new Date().setHours(23,59,59,0)),180);
    includeWeekends: boolean = false;
    daysRepeating: string = '1';
    weeksRepeating: string = '1';
    monthsRepeating: string = '1';
    weekendsIncluded: string = 'no';
    weekInterval: string = '1';
    weeklyRepeatType: string = 'number';
    monthlyRepeatType: string = 'number';
    monthlyDayType: string = 'number';

    constructor(recurrence?: RecurrenceFormValues){
        if(recurrence){
         this.sunday = recurrence.sunday;
         this.monday = recurrence.monday;
         this.tuesday = recurrence.tuesday;
         this.wednesday = recurrence.wednesday;
         this.thursday = recurrence.thursday;
         this.friday = recurrence.friday;
         this.saturday = recurrence.saturday;
         this.interval = recurrence.interval;
         this.dayOfMonth = recurrence.dayOfMonth;
         this.weekOfMonth = recurrence.weekOfMonth;
         this.weekdayOfMonth = recurrence.weekdayOfMonth;
         this.intervalStart = recurrence.intervalStart;
         this.intervalEnd = recurrence.intervalEnd;
         this.id = recurrence.id
         this.includeWeekends = recurrence.includeWeekends;
         this.daysRepeating = recurrence.daysRepeating;
         this.weekendsIncluded = recurrence.weekendsIncluded;
         this.weekInterval = recurrence.weekInterval;
         this.weeklyRepeatType = recurrence.weeklyRepeatType;
         this.weeksRepeating = recurrence.weeksRepeating;
         this.monthlyRepeatType = recurrence.monthlyRepeatType;
         this.monthlyDayType = recurrence.monthlyDayType;
         this.monthsRepeating = recurrence.monthsRepeating;
    }
   }
}